cli/Sources/TuistHasher/TargetScriptsContentHasher.swift

```
@@ -64,4 +84,63 @@ public struct TargetScriptsContentHasher: TargetScriptsContentHashing {
+            )
+        } else if fileListPath.extension == "xcfilelist" {
+            await processXCFileList(
+                fileListPath,
+                sourceRootPath: sourceRootPath,
+                pathsToHash: &pathsToHash,
+                stringsToHash: &stringsToHash
+            )
+        } else {
+            pathsToHash.append(fileListPath)
+        }
+    }
+
+    private func processXCFileList(
+        _ fileListPath: AbsolutePath,
+        sourceRootPath: AbsolutePath,
+        pathsToHash: inout [AbsolutePath],
+        stringsToHash: inout [String]
+    ) async {
+        do {
+            let fileListContent = try await fileSystem.readTextFile(at: fileListPath)
+            let pathStrings = fileListContent
+                .components(separatedBy: .newlines)
+                .map { $0.trimmingCharacters(in: .whitespaces) }
+                .filter { !$0.isEmpty && !$0.hasPrefix("#") }
+
+            for pathString in pathStrings {
+                if pathString.contains("$") {
+                    stringsToHash.append(pathString)
+                    Logger.current.notice(
+                        "The path '\(pathString)' is hashed, not the content. Because it has a build variable."
+                    )
+                } else if let absolutePath = try? AbsolutePath(validating: pathString) {
+                    pathsToHash.append(absolutePath)
+                } else {
+                    let resolvedPath = sourceRootPath.appending(try RelativePath(validating: pathString))
+                    pathsToHash.append(resolvedPath)
+                }
+            }
+        } catch {
+            Logger.current.warning(
+                "Failed to read .xcfilelist file at \'\(fileListPath.pathString)\': \(error). Hashing the file list path instead."
+            )
+            pathsToHash.append(fileListPath)
+        }
```

I'd bubble up the error